app/src/org/commcare/CommCareApplication.java

```diff
@@ -712,6 +712,7 @@ public void onServiceConnected(ComponentName className, IBinder service) {
                         }
                     }
 
+                    CommCareHeartbeatManager.instance().startHeartbeatCommunications();

```

Comment:
"The CommCareService already manages its lifecycle cleanly," --> But this `onServiceConnected` method is clearly responsible for initializing a lot of things that are supposed to be tied to the start of the session. What makes this different?